packages/shared/lib/services/sync/orchestrator.service.ts

```diff
@@ -404,6 +358,59 @@ export class Orchestrator {
             );
         }
     }
+    private async fetchSyncData(syncId: string, environmentId: number) {
+        const syncClient = await SyncClient.getInstance();
+        const schedule = await getSchedule(syncId);
+        const latestJob = await getLatestSyncJob(syncId);
+        let status = this.classifySyncStatus(latestJob?.status as SyncStatus, schedule?.status as ScheduleStatus);
+
+        const syncSchedule = await syncClient?.describeSchedule(schedule?.schedule_id as string);
+        if (syncSchedule) {
+            if (syncSchedule?.schedule?.state?.paused && status !== SyncStatus.PAUSED) {
+                await updateScheduleStatus(schedule?.id as string, SyncCommand.PAUSE, null, environmentId);
+                status = SyncStatus.PAUSED;
+            } else if (!syncSchedule?.schedule?.state?.paused && status === SyncStatus.PAUSED) {
+                await updateScheduleStatus(schedule?.id as string, SyncCommand.UNPAUSE, null, environmentId);
+                status = SyncStatus.STOPPED;
+            }
+        }
+
+        let nextScheduledSyncAt = null;
+        if (status !== SyncStatus.PAUSED) {
+            if (syncSchedule && syncSchedule?.info && syncSchedule?.info.futureActionTimes && syncSchedule?.info?.futureActionTimes?.length > 0) {
+                const futureRun = syncSchedule.info.futureActionTimes[0];
+                nextScheduledSyncAt = syncClient?.formatFutureRun(futureRun?.seconds?.toNumber() as number);
+            }
+        }
+
+        return { schedule, latestJob, status, nextScheduledSyncAt };
```

Comment:
Should we have some error handling in case any of the await fails?

Reply:
The call to temporal is in a try catch and the others are DB calls. The entire calling function is in a try, catch from the controller